components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java

```diff
@@ -220,6 +225,24 @@ public OMNode lookup(String key) {
         return result;
     }
 
+    private boolean lookupUtil(String key, URL url) {
+        if (url == null) {
+            handleException("Unable to create URL for target resource : " + key);
+        }
+
+        if ("file".equals(url.getProtocol())) {
+            try {
+                url.openStream();
+            } catch (IOException e) {
+                if (log.isDebugEnabled()) {
```

Comment:
Should this be an error log?

Reply:
This is just registry lookup and if the resource is not there we are returning null , hence I don't think this need to be an error log.

Reply:
but is it correct to say that an IOException would be thrown only for a missing file?

Reply:
added in 0ee8df54ff7fa587de65c6d09435573d5c8f59f2